tests/components/smartthings/test_init.py

```diff
@@ -22,12 +22,14 @@
     config_entry.version = 1
     setattr(hass.config_entries, '_entries', [config_entry])
     api = smartthings_mock.return_value
-    api.delete_installed_app.return_value = mock_coro()
+    api.delete_installed_app.side_effect = lambda _: mock_coro()
```

Comment:
Maybe use `mock_coro_func` from tests/common.py and set the return_value  for clarity. I'm guessing that has the same effect.

Reply:
Yeah -- I can't use it as a drop-in replacement to the mock because it doesn't track call count.  This line changed from `return_value` to `side_effect` because I found that you can't await the same instance of a `mock_coro` more than once.  If it does, it seemed to short-circuit all subsequent awaits too, even on non-mocked coroutines.  Side effect ensures a new instance is returned to be awaited should the call be invoked more than once.